frontend/src/components/ui/BuyMenu.jsx

```diff
@@ -1,19 +1,66 @@
+import { useState, useEffect } from "react";
 import { Text } from "@react-three/drei"
 import BuildGhostView from "../views/BuildGhostView";
 
-export default function BuyMenu({width, depth, towerConstructors, newTower, setNewTower}) {
+const NAME = "BuyMenu"
+
+export default function BuyMenu({game, towerConstructors}) {
+    const {gameMap, mouseInput} = game;
+    const {width, depth} = gameMap;
+
+    const [selected, setSelected] = useState(null);
+    const [newTower, setNewTower] = useState(null);
+
+    const deselectAll = () => {
+        setNewTower(null);
+        setSelected(null);
+        mouseInput.removeHoverCallback(NAME);
+        mouseInput.removeClickCallback(NAME);
+    };
+
+    useEffect(() => {
+        // Handle game state changes
+        deselectAll();
+    }, [game.phase])
+
     const exampleTowers = towerConstructors.map(constructor => new constructor());
 
     const buyButtons = exampleTowers.map((tower, i) => {
         const handleClick = () => {
             if (newTower?.name === tower.name) {
                 setNewTower(null);
             } else {
+                deselectAll();
                 const example = exampleTowers.find(example => tower.name === example.name);
-                if (example) setNewTower(new example.constructor());
+                if (example) {
+                    const t = new example.constructor();
+                    setNewTower(t);
+
+                    mouseInput.addHoverCallback(NAME, (x, y, z) => {
+                        t.setPosition(x, y, z);
+                    });
+
+                    const buildTower = (_x, _y, _z) => {
```

Comment:
Are the arguments here just to reinforce that it's not based on where the mouse clicks?

Reply:
That and also documenting that this function is being called with three args, even though it's not being used.

Reply:
Should we remove those args from the call?

Reply:
I think `MouseInput` is more complete with them in the call but it's a moot point if we never end up using it. It's a small change anyway, I'll remove them for now.